src/particles/collisions/UpdateMomentumPerez.H

```diff
@@ -88,6 +88,8 @@ void UpdateMomentumPerezElastic (
     T_R const vcy    = (p1y+p2y) / mass_g;
     T_R const vcz    = (p1z+p2z) / mass_g;
     T_R const vcms   = vcx*vcx + vcy*vcy + vcz*vcz;
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE(vcms*inv_c2_SI < 1,
+                "Particles with velocities > c in collisional module detected");
```

Comment:
Could you check that this doesn't have a significant performance impact? If so, I think we could take an upper bound here and/or in `vcx/y/z` (sth like `1.-std::numeric_limits<T_R>::epsilon` or so) provided that makes sense with the rest.